### ISSUE TITLE
read.odf(): do we need the 'NA coercion' warning?

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1183]
Below shows what I see. It doesn't happen very often, just 10 times out of 953 ODF files on my machine. Still, it's annoying, because it might make users think that something is quite wrong in the file (well, maybe something *is* quite wrong -- that's what this issue is about...)

```R
> f<-"/data/Barrow Strait Data/2005_2006/Bottom_Pressure/MTG_DGR2005001_1570_0282_3600.ODF"
> library(oce)
Loading required package: gsw
Loading required package: testthat
> options(warn=3)
> d<-read.oce(f)
Error in read.odf(file) : 
  (converted from warning) NAs introduced by coercion
> traceback()
6: doWithOneRestart(return(expr), restart)
5: withOneRestart(expr, restarts[[1L]])
4: withRestarts({
       .Internal(.signalCondition(simpleWarning(msg, call), msg, 
           call))
       .Internal(.dfltWarn(msg, call))
   }, muffleWarning = function() NULL)
3: .signalSimpleWarning("NAs introduced by coercion", quote(read.odf(file))) at odf.R#748
2: read.odf(file) at oce.R#1610
1: read.oce(f)
> 
```

**Comment by dankelley** [comment]
I see. This is in an ADCP file, and the null value is for time, viz.
```
  NULL_VALUE=18-NOV-1858 00:00:00.00,
```
and I think the real solution should be to change the way we work with null values, to allow a different one for each column. (This didn't seem worth the hassle for numeric-type files, but these ADCP files have time as a string.)

**Comment by dankelley** [comment]
Fixed in `develop` commit f20bb732b4ad08a5710927870a75d9bb1dd1249a but note that we only use the first numeric `NULL_VALUE` (and warn if we find character values or non-unique numeric values). I've put some comments in the file that will make it easy to find the relevant bits, e.g. if we find some cases where we actually need to use multiple `NULL_VALUE`s. I'm closing this, not as a perfect solution to handling `NA` values, but as a solution for the problem as stated. (Also, I should note, the present solution works fine for the approx. 1000 ODF files on my computer, as tested via `tests_private/odf*R`.)
